### ISSUE TITLE
Edit files without extension

### ISSUE THREAD

**Issue by eine** [filebrowser/filebrowser#314]
This issue is split from #302.

Currently double clicking is the only way to open the editor in order to modify the content of a file. However, it does not work if the file has no extension. Instead, it shows a black box covering all the page which suggests to download it. I think opening any file as text should be allowed.

Somehow related to this, I don't know if using double click to open folders and files is good for smartphones and other touch devices. Since there is already a 'Select multiple' activation option and both `Ctrl+Click` and `Shift+Click` work on desktops, I suggest making a single click open the file/folder. Moreover, this is coherent with default browsers' behaviour.

> [hacdias](https://github.com/hacdias/filemanager/issues/302#issuecomment-354128743)
> What if we try to open a binary file without extension?

There is no problem at all to open a binary file with a text edior. In the worst case you get non-understandable characters, e.g.:

```
E  èx"  …íÇ   …¶ýÿÿ1ÀH‡é¬ýÿÿH‰Ñÿ‡~  é&ýÿÿH‹E  H‹
ôD  Ç   è9"  émýÿÿE1íéæþÿÿ‰ÁèÍ!  @ Hƒì(H‹5E  Ç    èZ
  è¥üÿÿHƒÄ(Ãfffff.„     Hƒì(H‹E  Ç     è*
  èuüÿÿHƒÄ(ÃUH‰å]Ãf.„     UH‰åHƒì Hƒ= +   t0H
‡:  ÿ±}  H…Àt/Hƒ:  H‰Áÿ¤}  H…Àt	H
```

Moreover, some binary files use extensions such as `.bin`, `.run`, `.exec`... Therefore, I think it is better to not filter by name at all.

Shall JavaScript or golang provide some mechanism to identify if a file is going to be readable by a human, a filter can be applied based on that guess. The filter would show a warning letting the user know that the opened file is expected to be hardly readable.

**Comment by chinglinwen** [comment]
Really wanted this, ( say edit a Dockerfile, or any text file )

I also found it can't edit new create file ( from the ui and console, named a.go ),  I've seen some real go code file can be open and editing, don't know why this (say b.go ) can't ( echo aa > b.go )

**Comment by hacdias** [comment]
Are you suggesting something like #358? Could you both check if it serves your needs?

**Comment by eine** [comment]
@hacdias, it works fine for me!

Example:

```
git clone -b edit-everything https://github.com/filebrowser/filebrowser
cd filebrowser
docker build -t filebrowser/filebrowser .
docker run --rm -itv $(pwd):/srv -p 80:80 filebrowser/filebrowser --no-auth -p 80
```

Now `Dockerfile` can be edited.

---

Yet, you might want to make #358 optional, so that users willing to provide limited edition can configure it. I mean, I would not drop the existing feature.

**Comment by hacdias** [comment]
Hello! Will be available on v2.0! I'm closing this to remove the clutter in the meanwhile. For more info, check #574 and #575!
